src/server/api/controllers/waypoints.controller.js

```diff
@@ -0,0 +1,81 @@
+const HttpError = require("../lib/utils/http-error");
+const knex = require("../../config/db");
+
+// Method for getting waypoints by id:
+const getWaypointsById = async id => {
+  try {
+    const waypoints = await knex("waypoints")
+      .select("id", "lon", "lat", "created_at", "updated_at")
+      .where({ id: id });
+    if (waypoints.length === 0) {
+      throw new HttpError(
+        "Bad request",
+        `Cannot find waypoints for ID ${id}!`,
+        409
+      );
+    }
+    return waypoints;
+  } catch (err) {
+    return err.message;
+  }
+};
+
+// Method for creating a waypoints
+
+const createWaypoints = async ({ body }) => {
+  const { lat, lon, port_id, route_id } = body;
+
+  // 1. Get the route
+  const routes = await knex
+    .from("suggested_routes")
+    .select("*")
+    .where({ id: route_id });
+  if (routes.length === 0) {
+    throw new HttpError(
+      "Bad request",
+      `Cannot find routes for ID ${route_id}!`,
+      404
+    );
+  }
+
+  // 2. Get the port
+  const ports = await knex
+    .from("ports")
+    .select("*")
+    .where({ id: port_id });
+  if (ports.length === 0) {
+    throw new HttpError(
+      "Bad request",
+      `Cannot find ports for ID ${port_id}!`,
+      404
+    );
+  }
```

Comment:
We sort of discussed this already, but as soon as we get the port set to be nullable, we just set null on the port and this DB request is pointless :)